modules/3d/src/rgbd/normal.cpp

```diff
@@ -256,29 +239,60 @@ class RgbdNormalsImpl : public RgbdNormals
         CV_Assert(points3d_ori.dims == 2);
 
         // Either we have 3d points or a depth image
-        assertOnBadArg(points3d_ori);
+
+        bool ptsAre4F = (points3d_ori.channels() == 4) && (points3d_ori.depth() == CV_32F || points3d_ori.depth() == CV_64F);
+        bool ptsAreDepth = (points3d_ori.channels() == 1) && (points3d_ori.depth() == CV_16U || points3d_ori.depth() == CV_32F || points3d_ori.depth() == CV_64F);
+        if (method == RGBD_NORMALS_METHOD_FALS || method == RGBD_NORMALS_METHOD_SRI || method == RGBD_NORMALS_METHOD_CROSS_PRODUCT)
+        {
+            CV_Assert(ptsAre4F);
+        }
+        else if (method == RGBD_NORMALS_METHOD_LINEMOD)
+        {
+            CV_Assert(ptsAre4F || ptsAreDepth);
+        }
+        else
+        {
+            CV_Error(Error::StsInternal, "Unknown normal computer algorithm");
+        }
```

Comment:
Maybe it's better to change that bool's name to something more meaningful? Like `inputIs4floats`, `inputIs1channel`
Because guessing what the exact condition failed across the bool expression in `CV_Assert` can be irritating.
Or maybe replace it by `if(!condition) {CV_Error(..., "Meaningful explanation of what's expected ");}`